dali/base/dafdesc.cpp

```diff
@@ -3139,3 +3144,136 @@ void setupContainerizedStorageLocations()
         e->Release();
     }
 }
+
+static CriticalSection storageCS;
+void initializeStorageGroups(bool createPlanesFromGroups)
+{
+    CriticalBlock block(storageCS);
+    IPropertyTree * storage = queryGlobalConfig().queryPropTree("storage");
+    if (!storage)
+        storage = queryGlobalConfig().addPropTree("storage");
+
+#ifndef _CONTAINERIZED
+    if (createPlanesFromGroups && !storage->hasProp("planes[0]"))
+    {
+        //Create information about the storage planes from the groups published in dali
+        INamedGroupStore & groupStore = queryNamedGroupStore();
+        Owned<INamedGroupIterator> iter= groupStore.getIterator();
+        ForEach(*iter)
+        {
+            StringBuffer name, dir;
+            iter->get(name);
+            if (name.length())
+            {
+                IPropertyTree * plane = storage->addPropTree("planes");
+                plane->setProp("@name", name);
+                GroupType groupType;
+                Owned<IGroup> group = groupStore.lookup(name, dir, groupType);
+
+                if (dir.length())
+                    plane->setProp("@prefix", dir);
+                else
+                    plane->setProp("@prefix", queryBaseDirectory(groupType, 0));
+
+                if (!group)
+                    plane->setPropInt("@numDevices", 0);
+                else if (group->ordinality() != 1)
+                    plane->setPropInt("@numDevices", group->ordinality());
+
+                if (groupType == grp_thor)
+                {
+                    assertex(group);
+                    StringBuffer mirrorname;
+                    mirrorname.append(name).append("replica");
+
+                    IPropertyTree * mirror = storage->addPropTree("planes");
+                    mirror->setProp("@name", mirrorname);
+                    mirror->setProp("@prefix", queryBaseDirectory(groupType, 1));
+
+                    if (group->ordinality() != 1)
+                        mirror->setPropInt("@numDevices", group->ordinality());
+
+                    IPropertyTree * elem = plane->addPropTreeArrayItem("replication", createPTree());
+                    elem->setProp("", mirrorname);
+                }
+            }
+        }
+    }
+#endif
+
+    //Groups are case insensitve, so add an extra key to the storage items to allow them to be
+    //searched by group name, and also check for duplicates.
+    Owned<IPropertyTreeIterator> iter = storage->getElements("planes");
+    StringBuffer group;
+    ForEach(*iter)
+    {
+        IPropertyTree & cur = iter->query();
+        //Check if this has already been done - so the function is safe to call more than once
+        const char * oldgroup = cur.queryProp("@group");
+        if (oldgroup)
+            continue;
+
+        const char * name = cur.queryProp("@name");
+        group.clear().append(name).toLowerCase();
+
+#ifdef _DEBUG
```

Comment:
Why only check in debug mode?

Reply:
Is this more of a dbgassertex, or is it validating user-supplied info?

Reply:
I guess it is validating user input - checking there aren't storage planes that only different in their case.  I suspect the helm chart would fail to deploy in that case, but I don't think there is any harm in making this unconditional.  (Potential O(n^2) operation, but I can't see it ever being an issue, and Jake's hash change would limit it.)